### PR TITLE
Ai dev

### PR DESCRIPTION
**Pull request by MenesesGHZ**
Do actions logic partially finished

### CODE REVIEW

File: src/game/ai/agent.gd [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	sequence[0] = sequence[3]
+	sequence[1] = sequence[4]
+	sequence[2] = q_learning.compute_reward(state)
+	sequence[3] = q_learning.parse_state(state)
+	sequence[4] = action
+
+func valid_sequence() -> bool:
+	return sequence[0] != null and sequence[1] != null
+
+func do_action(action: String):
```
Comment (ClarkThyLord) [inline src/game/ai/agent.gd]: Check if Player is in dead state before doing action

File: src/game/player/player.gd [hunk 2]
```diff
@@ -80,31 +86,34 @@ func _process(delta : float) -> void:
 	
 	if ai_type == AiTypes.NONE:
 		var _player_type := "pong" if player_type == PlayerTypes.PONG else "ping"
```
Comment (ClarkThyLord) [inline src/game/player/player.gd]: We shouldn't remove user player logic, just append new logic as a subsection

**Review by ClarkThyLord — COMMENTED** [review]
AI is looking good, just a few bugs, that are: Agent is acting even when dead(e.g. shooting while in PlayerStates.DEAD). And we're also getting a bunch of errors with removing and adding Bullets. Other than that great job!

File: src/game/player/player.gd (hunk 2 above)
Comment (ClarkThyLord) [reply src/game/player/player.gd]: I'm just made some changes to check what type of AI the player has, however for whatever reason it's not being respected by the Q-learning agent, not sure if you can take a look?

File: src/game/ai/agent.gd (hunk 1 above)
Comment (MenesesGHZ) [reply src/game/ai/agent.gd]: Thanks man. Definitely there is a lot room from improvement, good catch, going to work on that.

File: src/game/player/player.gd (hunk 2 above)
Comment (MenesesGHZ) [reply src/game/player/player.gd]: checking

**Pull request merged**